hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/file/FileTableFunction.java

```diff
@@ -83,6 +94,43 @@ private static String randomName() {
         return SCHEMA_NAME_FILES + "_" + UuidUtil.newUnsecureUuidString().replace('-', '_');
     }
 
+    @Override
+    public List<Permission> permissions(SqlCall call) {
+        SqlNode astPath = findOperandByName(OPTION_PATH, call);
+        if (astPath == null) {
+            return Collections.emptyList();
```

Comment:
when we cannot validate permissions because something unexpected happened (lack of expected parameter or the parameter is of unexpected type) we might fail the query if it needs to check permissions (deny by default and allow only when we are sure that the action is allowed)

Reply:
https://github.com/hazelcast/hazelcast/pull/25784/commits/23a9065f45c9400a86631c4e99214e12ff96ea11